Formula/composer-switcher.rb

```diff
@@ -0,0 +1,18 @@
+class ComposerSwitcher < Formula
+  desc "Composer Mirror Repository Switcher"
+  homepage "https://www.github.com/persiliao/composer-switcher"
+  url "https://github.com/persiliao/composer-switcher/archive/v0.1.0.tar.gz"
+  version "0.1.0"
+  sha256 "8a2911dbdad2f6dfc036b6e55a5c79bc57aa1b35c6a5736ef6780b9f599e701a"
+
+  depends_on "composer"
+
+  def install
+    bin.install "composerswitcher.sh"
+    bin.install_symlink "composerswitcher.sh" => "composer-switcher"
+  end
+
+  test do
+    assert_match "usage: composer-switcher", shell_output("#{bin}/composer-switcher")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.